Adds quick start new visuals for quick start card

AjeshRPai opened the pull request:
Parent #16587 

This PR Updates the Quick Start card to new designs for the Existing Site tours (New Site tours are using the original designs)


## WordPress app

|Quick start Finished |  Quick start Finished(Dark mode)| 
|---|---|
|![Screenshot_20220527-150835.jpg](https://user-images.githubusercontent.com/17463767/170674331-e89d7d53-a6bd-43eb-9ecb-00d436a1b633.jpg)|![Screenshot_20220527-151337.jpg](https://user-images.githubusercontent.com/17463767/170675005-02b99c62-3de6-4bb7-bce6-0b3d8b74e101.jpg)|

---
### To test:

#### Existing Site tours

1. Login to an app with existing site tour quick start in progress 
3. Verify: The quick start card design has been updated
4. Complete a tour
5. Verify: The progress bar updates accordingly
6. Complete all tours
7. Verify: The progress bar turns green and a checkmark icon appears 

#### New Site tours
1. Login to an app with old quick start tour in progress
2. Verify: The quick start card design hasn't changed

### Merge Instructions 
- [x] Revert the commit [128c170](https://github.com/wordpress-mobile/WordPress-Android/pull/16629/commits/128c170883499d1dcffd6f8b1733f70e5e1fccb9). 
- [x] Remove not ready for merge label 
- Merge

## Regression Notes
1. Potential unintended areas of impact
 Quick start card not being shown for existing users and new users

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual tests 

4. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

AjeshRPai committed:
Adds: Tour illustration svg

AjeshRPai committed:
Add: QUick start card style

AjeshRPai committed:
Adds: Quick start progress drawable

AjeshRPai committed:
Adds: Quick start new card task type and card tool bar

AjeshRPai committed:
Merge branch 'trunk' into issue/16587-quick-start-new-visuals-for-quick-start-card

AjeshRPai committed:
Updates: Quick start rendering logic for new card

AjeshRPai committed:
Adds: All tasks completed string

AjeshRPai committed:
Adds: Title Icon to the quick start card

AjeshRPai committed:
Updates: Quick start builder logic and updates XML logic

AjeshRPai committed:
Adds: Logic for showing completed icon

AjeshRPai committed:
Updates: Progress and subtitle logic

AjeshRPai committed:
Updates: Quick start card source building logic

AjeshRPai committed:
Replaces: progress color logic with correct color

AjeshRPai committed:
Fixes: Lint issues

AjeshRPai committed:
Updates: naming to resolve lint error

AjeshRPai committed:
Delete checkstyle-idea.xml

AjeshRPai committed:
Reverts: Removed logic for testing

AjeshRPai committed:
Adds: Message to the RELEASE-NOTES

AjeshRPai committed:
Removes: Card dismissal logic for design review

AjeshRPai commented:
@ashiagr @osullivanchris 
As the issue #https://github.com/wordpress-mobile/WordPress-Android/issues/16615 exists, the quick start card will disappear as soon as the quick start is completed. In order to do the design review of the completed state of the quick start card, I have disabled the removal of the quick start card in the completed state with the commit [128c170](https://github.com/wordpress-mobile/WordPress-Android/pull/16629/commits/128c170883499d1dcffd6f8b1733f70e5e1fccb9). This will be reverted once the design review is done and @osullivanchris has approved the changes.

AjeshRPai committed:
Renames: Styles attribute

ashiagr commented on WordPress/src/main/res/layout/new_quick_start_task_type_item.xml:
```diff
@@ -0,0 +1,69 @@
+        app:layout_constraintHorizontal_bias="1.0"
+        app:layout_constraintTop_toTopOf="@+id/quick_start_item_subtitle" />
+
+    <ProgressBar
+        android:id="@+id/quick_start_item_progress"
+        style="?android:progressBarStyleHorizontal"
+        android:layout_width="0dp"
+        android:layout_height="4dp"
+        android:layout_marginTop="@dimen/margin_medium"
+        android:indeterminate="false"
+        android:progressDrawable="@drawable/quick_start_card_horizontal_progress_drawable"
+        app:layout_constraintEnd_toStartOf="@+id/quick_start_item_icon"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/quick_start_item_subtitle"
+        tools:progress="100"
+        tools:progressBackgroundTint="@color/primary_0"
+        tools:progressTint="@color/colorPrimary" />
```
Np: This part can be formatted.

AjeshRPai replied:
@ashiagr I have done the formatting in the commit. https://github.com/wordpress-mobile/WordPress-Android/pull/16629/commits/9782625feca58e29b926820934e929a33d627cbd.

ashiagr commented on WordPress/src/main/res/layout/new_quick_start_task_type_item.xml:
```diff
@@ -0,0 +1,69 @@
+        android:id="@+id/quick_start_item_subtitle"
+        style="@style/QuickStartTypeSubtitle"
+        android:layout_width="0dp"
+        android:layout_marginTop="4dp"
```
Np: Hardcoded margins can be replaced with values from `dimens.xml`.

AjeshRPai replied:
👍🏼 Done in https://github.com/wordpress-mobile/WordPress-Android/pull/16629/commits/9782625feca58e29b926820934e929a33d627cbd

ashiagr commented on WordPress/src/main/res/layout/new_quick_start_task_type_item.xml:
```diff
@@ -0,0 +1,69 @@
+        app:layout_constraintTop_toTopOf="parent" />
+
+    <ImageView
+        android:id="@+id/quick_start_item_icon"
```
Np: This illustration image is actually not an icon and can be renamed. Wdyt?

AjeshRPai replied:
👍🏼 I have renamed it to `quick_start_item_img` in commit https://github.com/wordpress-mobile/WordPress-Android/pull/16629/commits/9782625feca58e29b926820934e929a33d627cbd, let me know if you have any other suggestions. 🤔

ashiagr reviewed (commented):
Works as described!  Left a few nitpicks for you to consider.

Also, 
- Looks like you deleted `checkstyle-idea.xml.` Can you bring it back?
- Not sure what we decided about the icon on the card toolbar, and if it should be present for both `home` and `menu` tabs. I'll leave that discussion to you and Chris.

ashiagr commented:
👋  @AjeshRPai 

Two more things: 
1. We used the same illustration in this PR and in the qs tasks list modal new visuals available in `trunk` now. They are included twice with different names. Can we merge trunk to this PR and get rid of one of them?
2. Let's update the release note to include one item for quick start new visuals. We can include both PRs in it.

AjeshRPai committed:
Merge branch 'trunk' into issue/16587-quick-start-new-visuals-for-quick-start-card

AjeshRPai committed:
Updates: String message in release notes

AjeshRPai committed:
Removes: Duplicate resource and updates naming and usages

AjeshRPai commented:
Hey @ashiagr , Thanks for pointing out the changes

>We used the same illustration in this PR and in the qs tasks list modal new visuals available in trunk now. They are included twice with different names. Can we merge trunk to this PR and get rid of one of them?

Done in https://github.com/wordpress-mobile/WordPress-Android/pull/16629/commits/1d2dc8eaaf4db278b18b443d403bbe90a01ddf84
1. I removed both the files from your PR
2. renamed the image file to `ic_quick_start_tour_illustration` for better clarity
3. Update the usage in card and in quick start list 

>Let's update the release note to include one item for quick start new visuals. We can include both PRs in it.

Done in [962f145](https://github.com/wordpress-mobile/WordPress-Android/pull/16629/commits/962f1456ee8d8cfc5d53ab8ce3e7001097966c2d)

ashiagr commented:
Thanks @AjeshRPai!

> renamed the image file to ic_quick_start_tour_illustration for better clarity

Can we rename it to `img_quick_start_tour_illustration` as it is not an icon? 😅

AjeshRPai committed:
Updates: Naming of the image resource

AjeshRPai committed:
Updates: design implementation of the Get to know card

Removes: Redundant variables from quick start card data class
Adds: Card toolbar visibility parameter to the card
Adds: Ellipsis icon to the new quick start card
Updates: Quick start card rendering logic

AjeshRPai committed:
Merge branch 'trunk' into issue/16587-quick-start-new-visuals-for-quick-start-card

AjeshRPai committed:
Updates: Release notes

AjeshRPai committed:
Updates: the margin and padding of the quick start card

AjeshRPai commented:
@osullivanchris, I have a few questions  

Title 
- Does it need to be in two lines?. 
- Title string should be "Get to know the app"  or "Get to know the App"

Subtitle/Secondary Text(1 out of 6 Complete) 
- Suggestion: This looks a little bit more separated from the title with 16 dp margin-top, should we make it 8 dp?. 

Illustration 
- I have given the following size right now 
	- 54dp width hardcoded
	- height - top to the title top and bottom to the progress bar bottom
Is this fine?

Task completed Icon
- Can u share with me the size of the task completed tick mark shown. Right now the size is 24dpX24dp

❓  Is the design or specs of the card different in the home and menu tab?

AjeshRPai committed:
Fixes: Lint issues and updates binding variable

AjeshRPai committed:
Fixes: Ktlint error

AjeshRPai committed:
Reverts: checkstyle-idea.xml deleted

AjeshRPai committed:
Reverts: Checkstyle changes

AjeshRPai committed:
Update checkstyle-idea.xml

AjeshRPai committed:
Update checkstyle-idea.xml

AjeshRPai committed:
Removes: Checkstyle changes

AjeshRPai commented:
@ashiagr 

>Looks like you deleted checkstyle-idea.xml. Can you bring it back?

This was due to a setting in my IDE, sorry. I have reverted the deletion.

ashiagr commented:
Changes look good to me @AjeshRPai 🙇‍♀️ 
I'll approve the PR once https://github.com/wordpress-mobile/WordPress-Android/pull/16629#issuecomment-1138262671 is addressed. Good job.

osullivanchris reviewed (changes requested):
👋 Hey Ajesh. Thanks for the ping. The build is looking much better, and happy to answer the clarifying questions.

I have a couple of small details first:
- the heading text should be the same size as the 'Create your first/next post' heading text. Right now it looks much larger.
- The progress bar would look better 1dp larger in height
- Changing from the specs, I think it will look better if there's 32dp padding between the left content area (heading, subheading, progress bar) and the illustration on the right. At the moment the progress bar looks too close to the illustration. 

Now for your questions

> Does it need to be in two lines?.

No I don't think we would ever make a rule like that. It wouldn't be very good semantically. Although I do think the balance of the card is nice when it goes to two lines, we should let it fall whatever way it falls

> Title string should be "Get to know the app" or "Get to know the App"

Get to know the app

> Subtitle/Secondary Text(1 out of 6 Complete)...Suggestion: This looks a little bit more separated from the title with 16 dp margin-top, should we make it 8 dp?.

I see what you're saying but I think the space balances things vertically. Especially if 'Get to know the app' is only on one line on some devices, the content will be super short compared with the illustration. So I think we should keep 16. I also think its good for the progress bar and 'N out of N complete' to look more tight together than to the heading.

> Illustration size

54 width sounds right. Not sure I follow on the height. The original aspect ratio should be preserved so perhaps we should only define the width. For the vertical position, I was originally thinking centred to the content to the left. Maybe this is what you mean. Vertically centred between the title and the progress bar. 

Another thought I have on the height is now that we're seeing it looks quite different if the text wraps or not, whether the illustration could vary in size based on the content next to it. But that seems complex with low value.

> Task completed Icon

Oh yeah I never provided that sorry. I would reduce to 16dp. Thanks!

AjeshRPai committed:
Updates: design according to review comments

AjeshRPai commented:
Hey @osullivanchris, Thanks for the review. I have done the following changes now. 

- [x] The heading text should be the same size as the 'Create your first/next post' heading text.
- [x] The progress bar would look better 1dp larger in height
- [x] 32dp padding between the left content area (heading, subheading, progress bar) and the illustration on the right
- [x] Task completed Icon - Reduced to 16 dp 

Regarding your other replies 
>54 width sounds right. Not sure I follow on the height. The original aspect ratio should be preserved so perhaps we should only define the width. For the vertical position, I was originally thinking centred to the content to the left. Maybe this is what you mean

Yup, I have maintained the same aspect ratio and the illustration is centred to the content on the left. 👍🏼

AjeshRPai commented:
Hey @osullivanchris, In order to confirm 
❓ Is the design or specs of the card different in the home and menu tab?

AjeshRPai committed:
Updates: dimen usage in quick start icon

osullivanchris commented:
> ❓ Is the design or specs of the card different in the home and menu tab?

Sorry I missed that one! I think the screenshots of 'menu' are fine. I don't think we need to introduce two variations.

osullivanchris reviewed (changes requested):
Thanks @AjeshRPai this is looking really great. All the changes look excellent.

One final detail I just want to get spot on. It looks like you took my idea about scaling the illustration. Did you make it totally variable to the height of the container or something? 

Here's what I'm seeing. Though technically I think what you're doing makes sense, when the string is shorter, its just making the card look a bit small and boring. The illustration is just going a bit too small I feel, and thus the height of the card as well. HEre's how it looks on my device for example
![Screenshot_20220527-182310](https://user-images.githubusercontent.com/50576199/170761819-e50a5cc1-3aee-4372-8dc3-bd882948bc52.png)

I tried mocking up what I think looks right when there's a one line string or a two line string. There's two things I changed.

- On the left, the illustration is 40dp wide. On the right, its 56dp wide. I wonder if we could do the scaling like you have, but have a min or max value so that the card doesn't get too small or too large? Or maybe there's some other way do achieve this. 
- On the left (single line) the padding is increased to 24dp to balance the card. On the right its still 16dp. Another way to look at this would be to align the progress bar and subtitle to the bottom of the card vertically. But not sure if that is good practice.

Not sure how crazy these suggestions are. I do think this _looks right_, but I'm tweaking a lot just to make it look right in the case of one or two lines of text.
<img width="1884" alt="Screenshot 2022-05-27 at 18 35 32" src="https://user-images.githubusercontent.com/50576199/170762635-1610bace-9b85-4f12-a445-8d55b58a7298.png">

osullivanchris reviewed (approved):
I'm gonna change my comment to approve. As I don't want you to be blocked merging this on my nitpick changes. You can get it into the build and then do my tweaks after or something if preferred.

AjeshRPai committed:
Revert "Removes: Card dismissal logic for design review"

This reverts commit 128c1708

AjeshRPai commented:
@osullivanchris 

>On the left, the illustration is 40dp wide. On the right, its 56dp wide. I wonder if we could do the scaling like you have, but have a min or max value so that the card doesn't get too small or too large? Or maybe there's some other way do achieve this.

I think this can be done. Although I have some doubts regarding this implementation, I am going to take up this change in a separate PR and cc you in on it. 


@ashiagr As I have mentioned in the https://github.com/wordpress-mobile/WordPress-Android/pull/16629#issuecomment-1138262671, I have reverted the changes which were needed for the design review in the commit [285116d](https://github.com/wordpress-mobile/WordPress-Android/pull/16629/commits/285116de20a39d4932cae839c3bc6cd4b0753901). This PR is ready for another look. 👍🏼

osullivanchris commented:
> I think this can be done. Although I have some doubts regarding this implementation, I am going to take up this change in a separate PR and cc you in on it.

Revisiting this with fresh eyes. There's two possibilities

1. We're gonna have lots of cards with illustrations. Maybe its useful to have some kinda dynamic scaling for images which work across different content. I do like what you did with the scaling (assuming this is what's happening) but to get it right I feel we need a min, max value. But having that dependent on the content length sounds pretty complex.
2. To make things much simpler I should just find a reasonable size that works pretty well no matter the content length to the left of it. While 40/56 either looked too large or too small depending on the content; I tried 48dp width and it looks reasonable in both cases. See img below. 

Should we just do the latter and have a set 48dp width?

<img width="882" alt="Screenshot 2022-05-30 at 10 49 59" src="https://user-images.githubusercontent.com/50576199/170967561-638619b2-0636-4d23-aa1b-dc4ffd186683.png">

AjeshRPai commented:
@osullivanchris 

>but to get it right, I feel we need a min, max value. But having that dependent on the content length sounds pretty complex

Yup, If there is a value in it. We can try this approach. It would be complex to vary the width according to the content length and keep a min/max value. I am afraid we might run into another scenario where it would not look correct.  

> I tried 48dp width and it looks reasonable in both cases. Should we do the latter and have a set 48dp width?

Yup, this looks good, to be honest. I say we will go with it 👍🏼 . Shall I go ahead and implement ?

osullivanchris commented:
> Yup, this looks good, to be honest. I say we will go with it 👍🏼 . Shall I go ahead and implement ?

yeah lets do it! The value:effort of the other approach is questionable!
